packages/liveblocks-react-ui/src/components/internal/Attachment.tsx

```diff
@@ -0,0 +1,306 @@
+"use client";
+
+import type {
+  CommentAttachment,
+  CommentLocalAttachment,
+  CommentMixedAttachment,
+} from "@liveblocks/core";
+import { useAttachmentUrl, useIsInsideRoom } from "@liveblocks/react";
+import type {
+  ComponentPropsWithoutRef,
+  KeyboardEvent,
+  MouseEventHandler,
+  PointerEvent,
+} from "react";
+import React, { memo, useCallback, useEffect, useMemo, useState } from "react";
+
+import { CrossIcon } from "../../icons/Cross";
+import { SpinnerIcon } from "../../icons/Spinner";
+import { WarningIcon } from "../../icons/Warning";
+import type { GlobalOverrides } from "../../overrides";
+import { useOverrides } from "../../overrides";
+import { AttachmentTooLargeError } from "../../primitives";
+import { useComposerAttachmentsContextOrNull } from "../../primitives/Composer/contexts";
+import { classNames } from "../../utils/class-names";
+import { formatFileSize } from "../../utils/format-file-size";
+import { Tooltip } from "./Tooltip";
+
+interface FileAttachmentProps extends ComponentPropsWithoutRef<"div"> {
+  attachment: CommentMixedAttachment;
+  onDeleteClick?: MouseEventHandler<HTMLButtonElement>;
+  preventFocusOnDelete?: boolean;
+  overrides?: Partial<GlobalOverrides>;
+}
+
+const IMAGE_PREVIEW_MAX_SIZE = 50 * 1024 * 1024; // 50 MB
+
+const fileExtensionRegex = /^(.+?)(\.[^.]+)?$/;
+
+function splitFileName(name: string) {
+  const match = name.match(fileExtensionRegex);
+
+  return { base: match?.[1] ?? name, extension: match?.[2] };
+}
+
+function getFileAttachmentIconGlyph(mimeType: string) {
+  if (
+    mimeType === "application/zip" ||
+    mimeType === "application/x-rar-compressed" ||
+    mimeType === "application/x-7z-compressed" ||
+    mimeType === "application/x-zip-compressed" ||
+    mimeType === "application/x-tar" ||
+    mimeType === "application/gzip"
+  ) {
+    return (
+      <path d="M13 15h2v1h-1.5a.5.5 0 0 0 0 1H15v1h-1.5a.5.5 0 0 0 0 1H15v1h-1.5a.5.5 0 0 0 0 1h1a.5.5 0 0 0 .5-.5V20h1.5a.5.5 0 0 0 0-1H15v-1h1.5a.5.5 0 0 0 0-1H15v-1h1.5a.5.5 0 0 0 .5-.5V15a2 2 0 0 1 2 2v4a2 2 0 0 1-2 2h-4a2 2 0 0 1-2-2v-4a2 2 0 0 1 2-2Z" />
+    );
+  }
+
+  if (mimeType.startsWith("text/")) {
```

Comment:
```suggestion
  if (mimeType.startsWith("text/") || mimeType.startsWith("application/")) {
```

I prefer your text file icon to the blank file icon. Looking at the [list of mimetypes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types), most of the common `application` mimetypes are actually documents or text files, (e.,g. MS word, rtf, ebook, JSON).

Alternatively, just show the few above as text files, as especially MS word `.doc(x)` files are common.